pkg/kubectl/cmd/get.go

```diff
@@ -110,19 +116,23 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	if isWatch || isWatchOnly {
 		r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
 			NamespaceParam(cmdNamespace).DefaultNamespace().AllNamespaces(allNamespaces).
+			FilenameParam(enforceNamespace, filenames...).
 			SelectorParam(selector).
 			ResourceTypeOrNameArgs(true, args...).
 			SingleResourceType().
+			Latest().
 			Do()
 		if err != nil {
 			return err
 		}
 
-		mapping, err := r.ResourceMapping()
+		infos, err := r.Infos()
 		if err != nil {
 			return err
 		}
 
+		info := infos[0]
```

Comment:
Because Using the old way of retrieving mapping returned an error "a resource mapping could not be loaded from *resource.FileVisitor". I think we don't need to check the length of infos firstly, the precondition is  "length of infos is bigger than 0". Such as, "patch.go" and "func (r *Result) Watch(resourceVersion string) (watch.Interface, error)" don't check the length of infos too. What's your point?